packages/contracts-periphery/tasks/sync-drippie-config.ts

```diff
@@ -0,0 +1,64 @@
+import { task } from 'hardhat/config'
+import { LedgerSigner } from '@ethersproject/hardware-wallets'
+import { PopulatedTransaction } from 'ethers'
+
+import { getDrippieConfig } from '../src'
+
+task('sync-drippie-config').setAction(async (args, hre) => {
+  console.log(`connecting to ledger...`)
+  const signer = new LedgerSigner(
+    hre.ethers.provider,
+    'default',
+    hre.ethers.utils.defaultPath
+  )
+
+  console.log(`connecting to Drippie...`)
+  const Drippie = await hre.ethers.getContractAt(
+    'Drippie',
+    (
+      await hre.deployments.get('Drippie')
+    ).address,
+    signer
+  )
+
+  console.log(`loading Drippie config for network...`)
+  const config = await getDrippieConfig(hre)
+
+  // Need this to deal with annoying Ethers/Ledger 1559 issue.
+  const sendtx = async (tx: PopulatedTransaction): Promise<void> => {
+    const gas = await signer.estimateGas(tx)
+    tx.type = 1
+    tx.gasLimit = gas
+    const ret = await signer.sendTransaction(tx)
+    console.log(`sent tx ${ret.hash}...`)
+    await ret.wait()
+  }
+
+  console.log(`syncing Drippie config...`)
+  for (const [dripName, dripConfig] of Object.entries(config)) {
+    const drip = await Drippie.drips(dripName)
+    if (drip.status === 0) {
+      console.log(`creating drip ${dripName}...`)
+      await sendtx(
+        await Drippie.populateTransaction.create(dripName, dripConfig)
+      )
+    } else if (
+      drip.config.target.toLowerCase() !== dripConfig.target.toLowerCase() ||
```

Comment:
Perhaps modularizing this into a function with comments would make this easier to read/maintain longterm